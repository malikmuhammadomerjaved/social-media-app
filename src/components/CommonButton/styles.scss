@import "../../styles/commons.scss";

.common-button-wrapper {
  .common-button {
    @include flexView(center, center);

    background: linear-gradient(90deg, $gradient-from 3.25%, $gradient-to 100%);
    width: 100%;
    border-radius: 1rem;
    border: none;
    color: $white-color;
    font-size: 1.6rem;
    padding: 1.5rem 0;
    cursor: pointer;

    @media (min-width: $mobile-screen) {
      font-size: 2rem;
    }

    &:focus {
      outline: none;
    }
  }
}
