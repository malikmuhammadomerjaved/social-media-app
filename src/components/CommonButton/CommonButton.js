import { Link } from "react-router-dom";

import "./styles.scss";

const CommonButton = (props) => {
  const { label } = props;

  return (
    <article className="common-button-wrapper">
      <Link to="/event">
        <button className="common-button">{label}</button>
      </Link>
    </article>
  );
};

export default CommonButton;
