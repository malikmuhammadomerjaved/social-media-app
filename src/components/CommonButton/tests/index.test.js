import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import CommonButton from "../CommonButton";

describe("<CommonButton />", () => {
  const { container } = render(
    <BrowserRouter>
      <CommonButton />
    </BrowserRouter>
  );

  it("should match CommonButton component to snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});
