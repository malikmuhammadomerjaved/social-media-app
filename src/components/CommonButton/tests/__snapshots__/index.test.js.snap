// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommonButton /> should match CommonButton component to snapshot 1`] = `
<div>
  <article
    class="common-button-wrapper"
  >
    <a
      href="/event"
    >
      <button
        class="common-button"
      />
    </a>
  </article>
</div>
`;
